Pod/Classes/Workspace/KZNWorkspace.m

```diff
@@ -190,4 +192,146 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
   [tableView deselectRowAtIndexPath:indexPath animated:YES];
 }
 
-@end
\ No newline at end of file
+#pragma mark - Storage
+
+- (NSArray*)arrayWithNodesComposition
+{
+  NSMutableArray *objectsToStore = [NSMutableArray array];
+
+  for (int index = 0; index < self.nodes.count; index++) {
+    KZNNode *node = self.nodes[index];
+    NSString *nodeBaseClass = NSStringFromClass ([node class]);
+
+    NSMutableDictionary *currentNode = [NSMutableDictionary dictionary];
+    [currentNode setObject:[NSNumber numberWithInt:index] forKey:@"NodeIndex"];
```

Comment:
wouldn't it be better if we implemented `NSCoding` conformance on the Nodes ? then the whole archiving / dearchiving would be nicer and more maintainable?

Reply:
I tried before, but it was not possible due the nodes definition contains blocks. Blocks are not serializable.

Reply:
Ok, can you add constants for the keys as they are repeated in few places and it would be nicer to have them defined in a single place and reused

Reply:
Ok. I agree